File: jelly-vs-tower/python/tests/jelly_builder.py
from __future__ import annotations

from jelly_vs_tower import ColorType, Jelly


class JellyBuilder:
    def __init__(self) -> None:
        self._id = "jelly-1"
        self._color = ColorType.Blue
        self._health = 10

    def with_id(self, id: str) -> JellyBuilder:
        self._id = id
        return self

    def with_color(self, color: ColorType) -> JellyBuilder:
        self._color = color
        return self

    def with_health(self, health: int) -> JellyBuilder:
        self._health = health
        return self

    def build(self) -> Jelly:
        return Jelly(self._id, self._color, self._health)

File: jelly-vs-tower/typescript/tests/JellyBuilder.ts
import { ColorType } from '../src/ColorType.js';
import { Jelly } from '../src/Jelly.js';

export class JellyBuilder {
  private _id = 'jelly-1';
  private _color: ColorType = ColorType.Blue;
  private _health = 10;

  withId(id: string): this { this._id = id; return this; }
  withColor(color: ColorType): this { this._color = color; return this; }
  withHealth(health: number): this { this._health = health; return this; }

  build(): Jelly {
    return new Jelly(this._id, this._color, this._health);
  }
}
